### DISCUSSION TITLE
Parsing json

### DISCUSSION THREAD

**Discussion by GJena** [pydantic/pydantic#4266] [Question]
Hi, I have a dict that I am trying to convert to a Pydantic model. I have tried 2 approaches so far and it isn't providing me with the result I'd like. My expected output is **{"['test']": Sentiment(positive=7, negative=9)**

**Approach 1:**

models.py
```
class Sentiment(BaseModel):
    positive: int = 0
    negative: int = 0

class SentimentCollection(BaseModel):
    collection: Dict[str, List[SentimentType]]
```
test.py
```
input_dict = {"['test']": [{'negative': 9}, {'positive': 7}]}
output = parse_raw_as(Dict[str, List[Sentiment]], json.dumps(input_dict))
```
This gives an output of:
**{"['test']": [Sentiment(positive=0, negative=9), Sentiment(positive=7, negative=0)]**

**Approach 2:**

models.py
```
class Sentiment(BaseModel):
    positive: int = 0
    negative: int = 0

class SentimentCollection(BaseModel):
    collection: Dict[str, SentimentType]
```
test.py
```
input_dict = {"['test']": [{'negative': 9}, {'positive': 7}]}
output = parse_raw_as(Dict[str, Sentiment], json.dumps(input_dict))
```
This gives an output of:
**{"['test']": Sentiment(positive=0, negative=0)** where it seems to be taking the default values.

Any idea what would the right way to get all the values copied over in the same Sentiment object? Thank you

**Comment by samuelcolvin** [comment]
You can't have a list as the key to a dict as it's not hashable. That's a fundamental limitation of others. You can however use a dict.

Worth checking that what you're doing is valid python before thinking about pydantic.

**Reply by GJena** [reply]
The key of the dict is a string i.e. "['test']". Updated the code to clarify

**Reply by samuelcolvin** [reply]
That's not what you've said:

> My expected output is {['test']: Sentiment(positive=7, negative=9)

**Reply by GJena** [reply]
My bad, thank you for pointing it out - updated it. I'm trying to get the dict values into one Sentiment object, instead of having it split across multiple Sentiment objects, or having it overwritten by the default values

**Comment by GJena** [comment]
Modified input_dict to have value as a list of one dict, instead of a list of multiple dicts - this solved the issue

**Reply by samuelcolvin** [reply]
Great.
